components/server/src/auth/generic-auth-provider.ts

```diff
@@ -258,7 +267,17 @@ export abstract class GenericAuthProvider implements AuthProvider {
         });
 
         const isAlreadyLoggedIn = request.isAuthenticated() && User.is(request.user);
-        const authFlow = AuthFlow.get(request.session);
+
+        const state = request.query.state;
+        if (!state) {
+            log.error(cxt, `(${strategyName}) No state present on callback request.`, { clientInfo });
```

Comment:
This case is already handled in line 294ff, incl. a metric. Why handling it here again? Plus, it seems to shadow the `isAlreadyLoggedIn` case.

Reply:
This checks if the query contained state param, whereas 294 checks the auth flow exists (auth flow is derived (verified) out of the state param.

I can add the metirc here as well

Reply:
Ah, I now see where isAlreadyLoggedIn is coming from: From the DB, which i future won't make sense for. :+1:

Jep, having the metrics here would make sense I think.